src/formulas/parser.ts

```diff
@@ -171,54 +241,66 @@ function parseOperand(tokens: Token[]): AST {
   }
 }
 
-function parseFunctionArgs(tokens: Token[]): AST[] {
+function parseFunctionArgs(tokens: TokenList) {
   consumeOrThrow(tokens, "LEFT_PAREN", _t("Missing opening parenthesis"));
-  const nextToken = tokens[0];
+  const nextToken = tokens.current;
   if (nextToken?.type === "RIGHT_PAREN") {
-    consumeOrThrow(tokens, "RIGHT_PAREN");
-    return [];
+    const rightParen = consumeOrThrow(tokens, "RIGHT_PAREN");
+    return { args: [], rightParen };
   }
   const args: AST[] = [];
   args.push(parseOneFunctionArg(tokens));
-  while (tokens[0]?.type !== "RIGHT_PAREN") {
+  while (tokens.current?.type !== "RIGHT_PAREN") {
     consumeOrThrow(tokens, "ARG_SEPARATOR", _t("Wrong function call"));
     args.push(parseOneFunctionArg(tokens));
   }
-  consumeOrThrow(tokens, "RIGHT_PAREN");
-  return args;
+  const rightParen = consumeOrThrow(tokens, "RIGHT_PAREN");
+  return { args, rightParen };
 }
 
-function parseOneFunctionArg(tokens: Token[]): AST {
-  const nextToken = tokens[0];
+function parseOneFunctionArg(tokens: TokenList): AST {
+  const nextToken = tokens.current;
   if (nextToken?.type === "ARG_SEPARATOR" || nextToken?.type === "RIGHT_PAREN") {
     // arg is empty: "sum(1,,2)" "sum(,1)" "sum(1,)"
-    return { type: "EMPTY", value: "" };
+    return {
+      type: "EMPTY",
+      value: "",
+      tokenStartIndex: tokens.current!.tokenIndex,
+      tokenEndIndex: tokens.current!.tokenIndex,
+    };
   }
   return parseExpression(tokens);
 }
 
-function consumeOrThrow(tokens, type, message?) {
+function consumeOrThrow(tokens: TokenList, type, message?) {
```

Comment:
tiens, while you are updating this line, could you add the missing type?
```suggestion
function consumeOrThrow(tokens: TokenList, type, message?: string) {
```